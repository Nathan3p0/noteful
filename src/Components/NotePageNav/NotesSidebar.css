.activeNoteFolder {
    writing-mode: vertical-lr;
    text-orientation: upright;
    text-transform: uppercase;
    margin: 25px auto 0;
    width: 50px;
    font-size: xx-large;

}