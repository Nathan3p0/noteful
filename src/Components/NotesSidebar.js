import React from 'react'
import PropTypes from 'prop-types';

const NotesSidebar = (props) => {
    return ( 
        <nav>
            <button onClick={props.goBack}>Go Back</button>
            <h3>{props.folder}</h3>
        </nav>
     );
}

NotesSidebar.defaultProps = {
    goBack : () => {}
}

NotesSidebar.propTypes = {
    goBack : PropTypes.func.isRequired,
    folder : PropTypes.string.isRequired
}
 
export default NotesSidebar;