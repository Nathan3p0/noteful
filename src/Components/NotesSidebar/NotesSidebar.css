.noteNavBtnWrapper {
    max-width: 180px;
    margin: 0 0 0 auto;
}

.noteNavBtnWrapper button {
    margin: 10px auto 1rem;
    background-color: #8dba69;
    border: none;
    font-size: 1.1rem;
    width: 100%;
    font-weight: 500;
    color: #ffffff;
    padding: .5rem;
    text-transform: uppercase;
    border-radius: .2rem;
    transition: all .4s ease-in-out;
}

.noteNavBtnWrapper button:hover {
    background-color: #61ac27;
    border: none;
    color: white;
    box-shadow: .1rem .1rem rgba(0, 0, 0, 0.315);
}

.activeNoteFolder {
    writing-mode: vertical-lr;
    text-orientation: upright;
    text-transform: uppercase;
    margin: 0 0 0 auto;
    width: 75px;
    font-size: xx-large;
    color: #939393;
    height: 50vh;
}

