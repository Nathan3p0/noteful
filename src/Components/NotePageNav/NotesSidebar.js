import React from 'react'
import PropTypes from 'prop-types';
import './NotesSidebar.css';
import { NotefulContext } from '../NotefulContext';

const NotesSidebar = (props) => {

    const findNoteFolder = (noteId, folders, notes) => {
        console.log(noteId, notes, folders);
        const folderId = notes.find(note => note.id === noteId);
        return folders.find(folder => folder.id === folderId);
    }
    const noteId = props.match.params.noteid;
    return (
        <NotefulContext.Consumer>
            {({ folders, notes }) => (
                <React.Fragment>
                    <div className="noteNavBtnWrapper">
                        <button className="noteNavBtn" onClick={props.history.goBack}>Go Back</button>
                    </div>
                    <h3 className="activeNoteFolder">{console.log(findNoteFolder(noteId, folders, notes))}</h3>
                </React.Fragment>
            )}
        </NotefulContext.Consumer>
    );
}

NotesSidebar.defaultProps = {
    goBack: () => { },
    folder: ''
}

NotesSidebar.propTypes = {
    goBack: PropTypes.func.isRequired,
    folder: PropTypes.string.isRequired
}

export default NotesSidebar;