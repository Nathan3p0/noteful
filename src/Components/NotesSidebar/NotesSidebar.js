import React, { useContext, useEffect } from 'react'
import { useHistory } from 'react-router-dom';
import { NotefulContext } from '../NotefulContext';
import './NotesSidebar.css'

const NotesSidebar = (props) => {
    let history = useHistory();
    const value = useContext(NotefulContext);
    const handleGoBack = () => {
        history.goBack();
    }

    useEffect(() => {
        value.getCurrentNote(props.match.params.noteid)
    }, [])

    return (

        < React.Fragment >
            <div className="noteNavBtnWrapper">
                <button className="noteNavBtn" onClick={handleGoBack}>Go Back</button>
            </div>
            <h3 className="activeNoteFolder">{value.folderName.name}</h3>
        </React.Fragment >
    );
}

NotesSidebar.defaultProps = {
    folder: ''
}

export default NotesSidebar;