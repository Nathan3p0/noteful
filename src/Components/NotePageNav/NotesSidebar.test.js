import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NotesSidebar from './NotesSidebar'


describe('Notes Sidebar Component', () => {

    const sidebarProps = {
        goBack : () => {},
        folder: 'Spangley'
    }

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NotesSidebar />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty given no note info', () => {
        const tree = renderer.create(<NotesSidebar />).toJSON();
        expect(tree).toMatchSnapshot()
    })
    
    it('renders properly when given props', () => {
        const tree = renderer.create(<NotesSidebar goBack={sidebarProps.goBack} folder={sidebarProps.folder}/>).toJSON();
        expect(tree).toMatchSnapshot()
    })

})