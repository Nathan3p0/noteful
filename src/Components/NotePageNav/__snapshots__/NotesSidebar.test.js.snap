// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notes Sidebar Component renders empty given no note info 1`] = `
Array [
  <div
    className="noteNavBtnWrapper"
  >
    <button
      className="noteNavBtn"
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>,
  <h3
    className="activeNoteFolder"
  >
    
  </h3>,
]
`;

exports[`Notes Sidebar Component renders properly when given props 1`] = `
Array [
  <div
    className="noteNavBtnWrapper"
  >
    <button
      className="noteNavBtn"
      onClick={[Function]}
    >
      Go Back
    </button>
  </div>,
  <h3
    className="activeNoteFolder"
  >
    Spangley
  </h3>,
]
`;
